types/mysql__xdevapi/index.d.ts

```diff
@@ -0,0 +1,493 @@
+
+	bind(parameter: any): CollectionFind<T>;
+	bind(parameter: string, value: any): CollectionFind<T>;
+
+	sort(...SortExprStr: string[]): CollectionFind<T>;
+	sort(SortExprStr: string | ReadonlyArray<string>): CollectionFind<T>;
+
+	groupBy(...GroupByExprStr: string[]): CollectionFind<T>;
+	groupBy(GroupByExprStr: string | ReadonlyArray<string>): CollectionFind<T>;
+
+	having(expr: SearchConditionStr): CollectionFind<T>;
+
+	limit(count: number, offset?: number): CollectionFind<T>;
+
+	offset(value: number): CollectionFind<T>;
+
+	lockExclusive(mode: LockContention): CollectionFind<T>;
+
+	lockShared(mode: LockContention): CollectionFind<T>;
+}
+
+export class CollectionModify<T> implements Binding, CollectionOrdering, Limiting {
+	arrayAppend<
+		K extends keyof T = keyof T,
+		// tslint:disable-next-line:no-unnecessary-generics
+		J extends keyof T[K] = keyof T[K]
+		// tslint:disable-next-line:no-unnecessary-generics
+	>(field: K, any: T[K][J]): CollectionModify<T>;
+
+	arrayInsert<
+		K extends keyof T = keyof T,
+		// tslint:disable-next-line:no-unnecessary-generics
+		J extends keyof T[K] = keyof T[K]
+		// tslint:disable-next-line:no-unnecessary-generics
+	>(field: K, any: T[K][J]): CollectionModify<T>;
+
+	execute(): Promise<Result>;
+
+	getClassName(): string;
+
+	patch(properties: Partial<T>): CollectionModify<T>;
+
+	set<K extends keyof T = keyof T>(field: K, any: T[K]): CollectionModify<T>;
+
+	unset(fields: string | ReadonlyArray<string>): CollectionModify<T>;
+
+	bind(parameter: any): CollectionModify<T>;
+	bind(parameter: string, value: any): CollectionModify<T>;
+
+	sort(...SortExprStr: string[]): CollectionModify<T>;
+	sort(SortExprStr: string | ReadonlyArray<string>): CollectionModify<T>;
+
+	limit(count: number, offset?: number): CollectionModify<T>;
+
+	offset(value: number): CollectionModify<T>;
+}
+
+export class CollectionRemove<T> implements Binding, CollectionOrdering, Limiting {
+	execute(): Promise<Result>;
+
+	bind(parameter: any): CollectionRemove<T>;
+	bind(parameter: string, value: any): CollectionRemove<T>;
+
+	sort(...SortExprStr: string[]): CollectionRemove<T>;
+	sort(SortExprStr: string | ReadonlyArray<string>): CollectionRemove<T>;
+
+	limit(count: number, offset?: number): CollectionRemove<T>;
+
+	offset(value: number): CollectionRemove<T>;
+}
+
+export class Result {
+	getAffectedItemsCount(): number;
+
+	getAffectedRowsCount(): number;
+
+	getAutoIncrementValue(): number;
+
+	getGeneratedIds(): string[];
+
+	getWarnings(): Promise<Warning[]>;
+
+	getWarningsCount(): number;
+}
+
+export class Schema implements DatabaseObject {
+	// tslint:disable-next-line
+	createCollection<T>(
+		name: string,
+		options: CreateCollectionOptions
+	// tslint:disable-next-line
+	): Promise<Collection<T>>;
+
+	dropCollection(name: string): Promise<boolean>;
+
+	existsInDatabase(): Promise<boolean>;
+
+	getClassName(): string;
+
+	// tslint:disable-next-line
+	getCollection<T>(name: string): Collection<T>;
```

Comment:
As discussed [here](https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md), it's cleaner if this is just `getCollection(name: string): Collection<any>`.